themes/default/content/blog/using-pulumi-securely/index.md

```diff
@@ -0,0 +1,97 @@
+
+### Build Pulumi Teams
+
+[Teams]({{< relref "/docs/intro/pulumi-service/teams#creating-a-team" >}}) improve on basic role-based access control (RBAC) for administrators and users by allowing you to group users and their access to specific stacks in your organization.  For example, a team called `productionreadonly` could be created and scoped so that all members have only limited access to read the production stacks, while another team, `productionadmins`, would retain full privileges to those stacks.
+
+### Import Manually Created Resources
+
+Manually created cloud resources are error-prone and difficult to audit, iterate and improve on.  If you’ve created manual resources, consider [importing]({{< relref "/blog/changes-to-import" >}}) them to Pulumi programs.
+
+### Tag your Stacks
+
+We have previously covered [Tag Policies]({{< relref "/blog/automatically-enforcing-aws-resource-tagging-policies" >}}) as a best practice for the cloud resources themselves. Still, you can also tag your Pulumi stacks to help organize and visualize your application resources in logical groupings that will help you assess, respond and automate incident remediation. Here are two examples of using tagging your stacks automatically: in [Python](https://github.com/pulumi/examples/blob/master/aws-py-stackreference/team/__main__.py#L8-L13) and [TypeScript](https://github.com/pulumi/examples/blob/master/aws-ts-stackreference/team/index.ts#L17-L22).
+
+### Deploy from Pipelines
+
+Automate every step of your deployment process for anything beyond local development by creating ci/cd pipelines. Manual approval steps in your pipeline should be avoided. Consider creating automated checks to ensure your infrastructure is created only with approved [Pulumi Packages]({{< relref "/docs/guides/pulumi-packages" >}}) and automatically ensure your infrastructure is in compliance by using [CrossGuard]({{< relref "/docs/guides/crossguard" >}}).
+
+Leveraging [Automation API]({{< relref "/docs/guides/automation-api" >}}) as the programmatic interface for running Pulumi programs without the Pulumi CLI is a strongly typed and safe way to use Pulumi in embedded contexts such as web servers without having to shell out to a CLI. For example, [Elkjop Nordic uses Automation API to provide a Self Service Portal]({{< relref "/blog/how-elkjop-nordic-enables-developers-to-self-serve-infrastructure" >}}) to build a secure IT Vending Machine full of their application and infrastructure building blocks.
```

Comment:
Something about the phrase "uses Automation API to provide a portal to build a vending machine" that tripped me up. Is it that the portal "serves as a secure IT vending machine"?

```suggestion
Leveraging [Automation API]({{< relref "/docs/guides/automation-api" >}}) as the programmatic interface for running Pulumi programs without the Pulumi CLI is a strongly typed and safe way to use Pulumi in embedded contexts such as web servers without having to shell out to a CLI. For example, [Elkjop Nordic uses Automation API to provide a self-service portal]({{< relref "/blog/how-elkjop-nordic-enables-developers-to-self-serve-infrastructure" >}}) that serves as a secure IT vending machine full of their application and infrastructure building blocks.
```

Reply:
Updated